ga4gh/client.py

```diff
@@ -371,30 +389,71 @@ def searchReferences(
             request, "references", protocol.SearchReferencesResponse)
 
-    def searchCallSets(self, variantSetId, name=None):
+    def searchCallSets(self, variantSetId, name=None, bioSampleId=None):
         """
         Returns an iterator over the CallSets fulfilling the specified
         conditions from the specified VariantSet.
 
         :param str name: Only CallSets matching the specified name will
             be returned.
+        :param str bioSampleId: Returns only callsets with the given
+            Id.
         :return: An iterator over the :class:`ga4gh.protocol.CallSet`
             objects defined by the query parameters.
         """
         request = protocol.SearchCallSetsRequest()
         request.variantSetId = variantSetId
         request.name = name
+        request.bioSampleId = bioSampleId
         request.pageSize = self._pageSize
         return self._runSearchRequest(
             request, "callsets", protocol.SearchCallSetsResponse)
 
-    def searchReadGroupSets(self, datasetId, name=None):
+    def searchBioSamples(self, datasetId, name=None, individualId=None):
+        """
+        Returns an iterator over the BioSamples optionally matching
+        a name or individualId for the given datasetId.
+
+        :param str name: Only BioSamples given this name are returned
+        :param str individualId: Only return matches with this
+            individualId
+        :return: An iterator over the :class:`ga4gh.protocol.BioSample`
+            objects defined by the query parameters.
+        :rtype: iter
+        """
+        request = protocol.SearchBioSamplesRequest()
+        request.datasetId = datasetId
+        request.name = name
+        request.individualId = individualId
+        request.pageSize = self._pageSize
+        return self._runSearchRequest(
+            request, "biosamples", protocol.SearchBioSamplesResponse)
+
+    def searchIndividuals(self, datasetId, name=None):
+        """
+        Returns an iterator over the Individuals optionally matching
+        a name for the given datasetId.
+
+        :param str name: Only Individuals given this name are returned
+        :return: An iterator over the :class:`ga4gh.protocol.Individual`
+            objects defined by the query parameters.
+        :rtype: iter
+        """
+        request = protocol.SearchIndividualsRequest()
+        request.datasetId = datasetId
+        request.name = name
+        request.pageSize = self._pageSize
+        return self._runSearchRequest(
+            request, "individuals", protocol.SearchIndividualsResponse)
+
+    def searchReadGroupSets(self, datasetId, name=None, bioSampleId=None):
         """
         Returns an iterator over the ReadGroupSets fulfilling the specified
         conditions from the specified Dataset.
 
         :param str name: Only ReadGroupSets matching the specified name
             will be returned.
+        :param str bioSampleId: Filters readgroups according to the ID
```

Comment:
Follow the pattern here. "Only ReadGroupSets matching the specified BioSampleId will be returned."